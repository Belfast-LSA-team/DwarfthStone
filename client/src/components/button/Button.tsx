import React from "react";
import classnames from "classnames";

import "./button.css";

type ButtonProps = {
    className?: string;
    type: "button" | "submit" | "reset";
    style: "primary" | "secondary";
    stretch?: boolean;
    clickHandler: React.MouseEventHandler<HTMLButtonElement>;
    children: React.ReactNode;
};

export const Button = (props: ButtonProps) => {
    const { className, type, style, stretch, clickHandler, children } = props;
    const classNames = classnames("button", `button_${style}`, className, {
        button_fullwidth: stretch,
    });

    return (
        <button className={classNames} type={type} onClick={clickHandler}>
            {children}
        </button>
    );
};
