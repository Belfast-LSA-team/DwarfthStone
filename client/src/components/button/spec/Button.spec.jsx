import React from "react";
import { mount } from "enzyme";

import Button from "..";

describe("Button", () => {
    it("рендерится корректно", () => {
        const wrapper = mount(
            <Button
                type="button"
                style="primary"
                stretch={false}
                clickHandler={() => {}}
            >
                Test Button
            </Button>
        );

        expect(wrapper).toMatchSnapshot();
    });
});
