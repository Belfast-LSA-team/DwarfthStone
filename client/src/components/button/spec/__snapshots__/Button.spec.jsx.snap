// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button рендерится корректно 1`] = `
<Button
  clickHandler={[Function]}
  stretch={false}
  style="primary"
  type="button"
>
  <button
    className="button button_primary"
    onClick={[Function]}
    type="button"
  >
    Test Button
  </button>
</Button>
`;
